origins_content_issue/src/Form/SettingsForm.php

```diff
@@ -0,0 +1,242 @@
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Config\TypedConfigManagerInterface;
+use Drupal\Core\File\FileExists;
+use Drupal\Core\File\FileSystemInterface;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Configure Origins content issue settings for this site.
+ */
+final class SettingsForm extends ConfigFormBase {
+
+  /**
+   * The filesystem path to the module.
+   */
+  private string $modulePath;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId(): string {
+    return 'origins_content_issue_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames(): array {
+    return ['origins_content_issue.settings'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    $this->modulePath = \Drupal::service('module_handler')->getModule('origins_content_issue')->getPath();
+
+    $form['notifications'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Notifications'),
+      '#open' => TRUE,
+    ];
+
+    $fields = \Drupal::service('entity_field.manager')->getFieldDefinitions('content_issue', 'content_issue');
+    $state_values = $fields['state']->getSetting('allowed_values');
+
+    $form['notifications']['issues'] = [
+      '#type' => 'html_tag',
+      '#tag' => 'strong',
+      '#value' => $this->t('Content issues'),
+    ];
+
+    $form['notifications']['notify_on_create'] = [
+      '#type' => 'checkbox',
+      '#description' => $this->t('Email assigned to when a new issue is created.'),
+      '#default_value' => (boolean) $this->config('origins_content_issue.settings')->get('notify_on_create') ?? TRUE,
+      '#wrapper_attributes' => ['class' => ['container-inline']],
+    ];
+
+    $notify_reporter = $this->config('origins_content_issue.settings')->get('notify_reporter') ?? '';
+
+    $form['notifications']['notify_state_change_to_reporter'] = [
+      '#type' => 'checkboxes',
+      '#description' => $this->t('Email issue reporter when the issue state is changed.'),
+      '#options' => $state_values,
+      '#default_value' => explode(',', $notify_reporter),
+      '#wrapper_attributes' => ['class' => ['container-inline']],
+    ];
+
+    $form['notifications']['comments'] = [
+      '#type' => 'html_tag',
+      '#tag' => 'strong',
+      '#value' => $this->t('Content issue comments'),
+    ];
+
+    $form['notifications']['notify_on_comment'] = [
+      '#type' => 'checkbox',
+      '#description' => $this->t('Email assigned to and reported by when comments are created or edited.'),
+      '#default_value' => (boolean) $this->config('origins_content_issue.settings')->get('notify_on_comment') ?? TRUE,
+      '#wrapper_attributes' => ['class' => ['container-inline']],
+    ];
+
+    $selected_icon = $this->config('origins_content_issue.settings')->get('report_icon') ?? 'icon_report_1.png';
+
+    for ($i = 1; $i <= 6; $i++) {
```

Comment:
6 looks like a magic number here. Better to set a const or comment what the 6 refers to?

Reply:
Put that down to WiP ;)